include/aos/common/tools/fs.hpp

```diff
@@ -348,6 +348,38 @@ class FS {
 #endif
         return ErrorEnum::eNone;
     }
+
+    /**
+     * Reads content of the file named by fileName into the given string.
+     *
+     * @param fileName file name.
+     * @param[out] text result string.
+     * @return Error.
+     */
+    static Error ReadFileToString(const String& fileName, String& text)
+    {
```

Comment:
restore the discussion:

@[al1img](https://github.com/al1img) al1img [Dec 15, 2023](https://github.com/aoscloud/aos_core_lib_cpp/pull/47#discussion_r1428076393)

Your C++ example more related how stream handle lines in text file rather than deal with string. How do you read whole file into std::string with one method in C++?

Also here is golang API:

func ReadFile(name string) []byte

My consern here, you can't use this API for reading binary files in case we need it in the future. And we will need to implement separate methods for such case. Having byte array covers both cases and we may use one implementation for both.

But if you really like to read direct to string, then these methods should be renamed to: ReadFileToString and WriteStringToFile accordingly.
Author
@[mykola-kobets-epam](https://github.com/mykola-kobets-epam) mykola-kobets-epam [Dec 15, 2023](https://github.com/aoscloud/aos_core_lib_cpp/pull/47#discussion_r1428160966)

    How do you read whole file into std::string with one method in C++?

One example I have found:
string get_file_string(){
std::ifstream ifs("path_to_file");
return string((std::istreambuf_iterator(ifs)), (std::istreambuf_iterator()));
}

    Having byte array covers both cases and we may use one implementation for both.

As I see golang has single readFile method to read content into a byte array. And conversion constructor to convert byte array into a string. So basically we need two methods as well.

There are similary apis in other languages. example for java:
Java 7
String content = new String(Files.readAllBytes(Paths.get("readMe.txt")), StandardCharsets.UTF_8);
Java 11
String content = Files.readString(Paths.get("readMe.txt"));

I chose the latest approach as it is more convenient,
otherwise it would be needed to allocate additional byte array, read content there & make conversion to string.

    these methods should be renamed to: ReadFileToString and WriteStringToFile accordingly.

ok, I will rename them.

Reply:
In all provided by you examples, byte array converted to string.